src/wanderer.ts

```diff
@@ -46,165 +52,200 @@ export function digitizedMonstersRemaining(): number {
+    noInv: true,
   },
 ];
 
-function acceptBestGuzzlrQuest() {
-  if (!Guzzlr.isQuestActive()) {
-    if (
-      Guzzlr.canPlatinum() &&
-      (!Guzzlr.haveFullPlatinumBonus() ||
-        (Guzzlr.haveFullBronzeBonus() && Guzzlr.haveFullGoldBonus()))
-    ) {
-      Guzzlr.acceptPlatinum();
-    } else if (Guzzlr.canGold() && (!Guzzlr.haveFullGoldBonus() || Guzzlr.haveFullBronzeBonus())) {
-      Guzzlr.acceptGold();
-    } else {
-      Guzzlr.acceptBronze();
-    }
-  }
-}
+function testZoneAndUsePotionToAccess(zone: Location | null) {
+  if (!zone) return false;
 
-function testZoneAndUsePotionToAccess() {
-  const guzzlZone = Guzzlr.getLocation();
-  if (!guzzlZone) return false;
-  const forbiddenZones: string[] = [""]; //can't stockpile these potions,
-  if (!get("_spookyAirportToday") && !get("spookyAirportAlways")) {
-    forbiddenZones.push("Conspiracy Island");
-  }
-  if (!get("_stenchAirportToday") && !get("stenchAirportAlways")) {
-    forbiddenZones.push("Dinseylandfill");
-  }
-  if (!get("_hotAirportToday") && !get("hotAirportAlways")) {
-    forbiddenZones.push("That 70s Volcano");
-  }
-  if (!get("_coldAirportToday") && !get("coldAirportAlways")) {
-    forbiddenZones.push("The Glaciest");
-  }
-  if (!get("_sleazeAirportToday") && !get("sleazeAirportAlways")) {
-    forbiddenZones.push("Spring Break Beach");
+  // these zones have equip or effect requirements that we don't want to deal with
+  const skiplist = $locations`The Oasis, The Bubblin' Caldera, Barrrney's Barrr, The F'c'le, The Poop Deck, Belowdecks, 8-Bit Realm, Madness Bakery, The Secret Government Laboratory`;
+  if (skiplist.includes(zone)) {
+    return false;
   }
-
-  zonePotions.forEach((place) => {
-    if (guzzlZone.zone === place.zone && !have(place.effect)) {
-      if (!have(place.potion)) {
-        buy(1, place.potion, 10000);
+  zoneUnlockers.forEach((place) => {
+    if (zone.zone === place.zone && !place.available()) {
+      if (
+        (!place.noInv && have(place.unlocker)) ||
+        buy(1, place.unlocker, WANDERER_PRICE_THRESHOLD) > 0
+      ) {
+        use(1, place.unlocker);
       }
-      use(1, place.potion);
     }
   });
-  const skiplist = $locations`The Oasis, The Bubblin' Caldera, Barrrney's Barrr, The F'c'le, The Poop Deck, Belowdecks, 8-Bit Realm, Madness Bakery, The Secret Government Laboratory`;
-  if (
-    forbiddenZones.includes(guzzlZone.zone) ||
-    skiplist.includes(guzzlZone) ||
-    guzzlZone.environment === "underwater" ||
-    !canAdv(guzzlZone, false) ||
-    (guzzlZone === $location`The Upper Chamber` && questStep("questL11Pyramid") === -1) // (hopefully) temporary fix for canadv bug that results in infinite loop
-  ) {
+  const canAdvHack = zone === $location`The Upper Chamber` && questStep("questL11Pyramid") === -1; // (hopefully) temporary fix for canadv bug that results in infinite loop
+  if (zone.environment === "underwater" || canAdvHack || !canAdv(zone, false)) {
```

Comment:
again, pretty sure I made this, but can't we just return `!(zone.environment === "underwater" || ...)`? Rather than `if (condition) return false; else return true;` couldn't we just do `return !condition`? is there a stylistic reason to choose this over that?